File: app/src/test/java/com/babestudios/companyinfouk/ui/favourites/FavouritesPresenterTest.java
package com.babestudios.companyinfouk.ui.favourites;

import com.babestudios.companyinfouk.data.DataManager;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Matchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class FavouritesPresenterTest {


	FavouritesPresenter favouritesPresenter;

	@Before
	public void setUp() {
		favouritesPresenter = new FavouritesPresenter(mock(DataManager.class));
		favouritesPresenter.create();
		FavouritesActivityView view = mock(FavouritesActivityView.class);
		favouritesPresenter.attachView(view);
	}

	@Test
	public void onAttachView_shouldCallShowFavouritesOnView() {
		verify(favouritesPresenter.getView()).showFavourites(any());
	}

	@Test
	public void whenGetCompany_shouldCallStartCompanyActivityOnView() {
		favouritesPresenter.getCompany(any(), any());
		verify(favouritesPresenter.getView()).startCompanyActivity(any(), any());
	}

	@Test
	public void whenRemoveFavourite_shouldCallDataManagerRemoveFavourite() {
		favouritesPresenter.removeFavourite(any());
		verify(favouritesPresenter.dataManager).removeFavourite(any());
	}
}

File: app/src/test/java/com/babestudios/companyinfouk/ui/favourites/FavouritesPresenterTest.kt
package com.babestudios.companyinfouk.ui.favourites

/*
@RunWith(MockitoJUnitRunner::class)
class FavouritesPresenterTest {


	private lateinit var favouritesPresenter: FavouritesPresenter

	@Before
	fun setUp() {
		val mockCompaniesRepository: CompaniesRepository = mock()
		val view: FavouritesActivityView = mock()
		favouritesPresenter = FavouritesPresenter(mockCompaniesRepository)
		favouritesPresenter.create()
		`when`<Array<SearchHistoryItem>>(mockCompaniesRepository.favourites).thenReturn(arrayOf(SearchHistoryItem("", "", 0L)))
		favouritesPresenter.attachView(view)
	}

	@Test
	fun onAttachView_shouldCallShowFavouritesOnView() {
		verify(favouritesPresenter.view!!).showFavourites(any())
	}

	@Test
	fun whenGetCompany_shouldCallStartCompanyActivityOnView() {
		favouritesPresenter.getCompany("", "")
		verify(favouritesPresenter.view!!).startCompanyActivity(any(), any())
	}

	@Test
	fun whenRemoveFavourite_shouldCallDataManagerRemoveFavourite() {
		favouritesPresenter.removeFavourite(SearchHistoryItem("", "", 0L))
		verify(favouritesPresenter.companiesRepository).removeFavourite(any())
	}
}
*/